go/parquet/internal/testutils/random_arrow.go

```diff
@@ -0,0 +1,475 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+	"golang.org/x/exp/rand"
+)
+
+func RandomNonNull(dt arrow.DataType, size int) array.Interface {
```

Comment:
based on the name of the files, Arrow might be the better package here?

Reply:
so these are actually test utilities that i end up using for generating random arrow arrays for use with the unit tests for writing arrow directly to parquet inside of a `pqarrow` package that isn't included here. As mentioned above, by putting it in the `internal` package path, these cannot be imported outside of the parquet packages. If i put this into the Arrow package, then I'd need to make them exported publicly in the arrow package which i was trying to avoid doing.

Reply:
this seems OK for now.  It seems like the best solution would be having a "test utility package" that both Arrow and Parquet could share for tests.